cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -292,20 +304,25 @@ func registryUrlFor(cluster string) string {
 	}
 }
 
-func appendRegistrySecretItemContext(c *secretbootstrap.Config, name string, cluster string, value secretbootstrap.DockerConfigJSONData) error {
+func updateRegistrySecretItemContext(c *secretbootstrap.Config, name, cluster string, value secretbootstrap.DockerConfigJSONData) error {
 	logrus.Infof("Appending registry secret item to: {name: %s, cluster: %s}", name, cluster)
-	sc, err := findSecretConfig(name, cluster, c.Secrets)
+	_, sc, err := findSecretConfig(name, cluster, c.Secrets)
 	if err != nil {
 		return err
 	}
+	for _, si := range sc.From[dotDockerConfigJson].DockerConfigJSONData {
+		if si.AuthField == value.AuthField && si.RegistryURL == value.RegistryURL {
+			return nil
+		}
+	}
 	sc.From[dotDockerConfigJson] = secretbootstrap.ItemContext{
 		DockerConfigJSONData: append(sc.From[dotDockerConfigJson].DockerConfigJSONData, value),
 	}
 	return nil
 }
 
 func updateBuildFarmSecrets(c *secretbootstrap.Config, o options) error {
-	buildFarmCredentials, err := findSecretConfig(fmt.Sprintf("%s-%s", buildFarm, credentials), string(api.ClusterAPPCI), c.Secrets)
+	_, buildFarmCredentials, err := findSecretConfig(fmt.Sprintf("%s-%s", buildFarm, credentials), string(api.ClusterAPPCI), c.Secrets)
```

Comment:
ATM, we need this only for b01, b02, vsphere, i.e., ensure they are there in case of `update`.
For the new clusters, it depends on how we implement https://issues.redhat.com/browse/DPTP-2497.
As far as I can see, we will do Dex as the only IDP.
We can discuss the details via Slack too.

Reply:
So you are saying we only want to do this for `build01`,`build02`, and `vsphere`?

Reply:
Yes, it is correct.
Github IDP wont be set up on the new clusters.